libs/utils/include/celix_convert_utils.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CELIX_CELIX_CONVERT_UTILS_H
+#define CELIX_CELIX_CONVERT_UTILS_H
+
+#include <stdbool.h>
+#include "celix_version.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file celix_convert_utils.h
+ * @brief The celix_convert_utils.h file contains utility functions to convert strings to other types.
+ */
+
+/**
+ * @brief Convert a string to a boolean.
+ *
+ * @param[in] val The string to convert.
```

Comment:
Value needs to be in 32 characters, else default is returned

Reply:
update impl to use celix_utils_writeOrCreateString